modules/generic-scala3/src/main/scala/pureconfig/generic/scala3/HintsAwareConfigReaderDerivation.scala

```diff
@@ -0,0 +1,31 @@
+package pureconfig
+package generic
+package scala3
+
+import scala.compiletime._
+import scala.deriving.Mirror
+
+trait HintsAwareConfigReaderDerivation
+    extends HintsAwareCoproductConfigReaderDerivation,
+      HintsAwareProductConfigReaderDerivation {
+  inline def deriveReader[A](using m: Mirror.Of[A], ph: ProductHint[A], cph: CoproductHint[A]): ConfigReader[A] =
+    inline m match {
+      case pm: Mirror.ProductOf[A] => deriveProductReader[A](using pm, ph)
+      case sm: Mirror.SumOf[A] => deriveSumReader[A](using sm, cph)
```

Comment:
After requiring only one type of hint in each method I had to pass arguments explicitly here. Otherwise I was getting compiler error of `wrong number of arguments at posttyper`.